### PR TITLE
SQL-2384: ODBC Documentation

### PR DESCRIPTION
**Pull request by terakilobyte**
And here it is.

### CODE REVIEW

File: evergreen.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@ command_type: system
 # 12 minutes is the longest we'll ever run
 exec_timeout_secs: 3600 # 12 minutes is the longest we'll ever run
 
+ignore:
+  - "*.md"
```
Comment (terakilobyte) [inline evergreen.yml:18]: This is the only substantive change to this file. The rest were just auto formatting changes when I saved ✊

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,844 @@
...
+ODBC Driver.
+5. Enter your connection information. At a minimum, you must enter:
+- DSN: A name for your new DSN.
+- Username: A database username to use to connect to your database.
+- Password: The database user's password.
+- MongoDB URI: Your MongoDB deployment URI.
+- Database: The name of the database to which to connect.
+- Enable maximum: Checkbox to enforce maximum string length of 4000 characters. You
+must enable this option to work with BI tools like Microsoft SQL Sever Management Studio that
+can't upport variable length string data with unknown maximum length.
```
Comment (bucaojit) [inline docs/overview.md:119]:
```suggestion
can't support variable length string data with unknown maximum length.
```

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,844 @@
...
+| Connected!                            |
+|                                       |
+| sql-statement                         |
+| help [tablename]                      |
+| quit                                  |
+|                                       |
++---------------------------------------+
+```
+
+Note: The warning `[MongoDB][API] Bufer size "0" not large enough for data.` does not
```
Comment (bucaojit) [inline docs/overview.md:187]:
```suggestion
Note: The warning `[MongoDB][API] Buffer size "0" not large enough for data.` does not
```

File: docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,844 @@
...
+By default, the MongoDB Atlas SQL ODBC driver produces logs in `/%HOME%/MongoDB/Atlas SQL/logs`.
+
+Logging can be fine-tuned by passing the `LOGLEVEL` property in your ODBC connection
+string or configuring it in your DSN.
+
+The following is a list of valid values for LOGLEVEL and their precedence:
+- ERROR - Only errors will be logged.
+- WARN - Information about operations that could be an error in future versions of the driver.
+- INFO - Informational log messages. Default
+- DEBUG - Debug information useful for debegging purposes. Enable this mode to submit a log with a HELP ticket.
```
Comment (bucaojit) [inline docs/overview.md:800]:
```suggestion
- DEBUG - Debug information useful for debugging purposes. Enable this mode to submit a log with a HELP ticket.
```

File: docs/overview.md [hunk 5]
```diff
@@ -0,0 +1,844 @@
...
+Key Features: Native object and array support.
+
+Compatibility: The MongoDB Atlas SQL ODBC driver is compatible with MongoDB versions greater
+than or equal to 6.0.6.
+
+## System Requirements
+
+### Hardware Requirements
+
+The MongoDB Atlas SQL ODBC driver has no hard system requirements itself. Any system
```
Comment (bucaojit) [inline docs/overview.md:19]:
```suggestion
The MongoDB Atlas SQL ODBC driver has no hardware system requirements itself. Any system
```

**Review by bucaojit — COMMENTED** [review]
Small corrections for now as I'm reading through, I'll review the content this afternoon.

**Review by alexiantonino — APPROVED** [review]

File: docs/overview.md (hunk 5 above)
Comment (bucaojit) [reply docs/overview.md:19]: confirmed `hard` was the intentional wording

File: docs/overview.md [hunk 6]
```diff
@@ -0,0 +1,844 @@
...
+capable of running a modern SQL tool (such as Power BI, DBeaver) is capable of running
+the MongoDB Atlas SQL ODBC driver.
+
+### Software Requirements
+
+#### Supported Operating Systems
+
+The MongoDB ODBC driver is compatible with Windows x86_64 architecture, linux x86_64,
+and linux arm2 systems.
```
Comment (bucaojit) [inline docs/overview.md:28]: What do you mean by `arm2 systems`?  I usually see arm64 or maybe amazon linux 2 (ARM)

File: docs/overview.md [hunk 7]
```diff
@@ -0,0 +1,844 @@
...
+
+#### Dependencies
+
+When querying a MongoDB standalone server or cluster (not [Atlas SQL](https://www.mongodb.com/docs/atlas/data-federation/query/query-with-sql/) powered by [Atlas Data Federation](https://www.mongodb.com/docs/atlas/data-federation/)), the accompanying
+`mongosqltranslate.dll` or `libmongosqltranslate.a` library is required to be colocated with the driver (`mongoodbc.dll` or `libatsql.so`).
+
```
Comment (bucaojit) [inline docs/overview.md:34]: Should we put the common location for this? ie
For Power BI users, the default installation path of the driver is:
```
C:\Program Files\MongoDB\Atlas SQL ODBC Driver\bin\
```
Although, I'm not sure if we're going to change that soon when the naming is changed.

File: docs/overview.md [hunk 8]
```diff
@@ -0,0 +1,844 @@
...
+SELECT
+  b.ProductSold,
+  CAST(b._id AS string) AS ID,
+  (b.Price *b.Quantity) AS totalAmount
+FROM
+  (SELECT* FROM Sales a WHERE customer.gender = 'F') a
+INNER JOIN
+  Transactions b
+ON
+  (CAST(a._id AS string) = CAST(b._id AS string));
```
Comment (bucaojit) [inline docs/overview.md:347]:
```suggestion
SELECT
  b.ProductSold,
  CAST(b._id AS string) AS ID,
  (b.Price * b.Quantity) AS totalAmount
FROM
  (SELECT * FROM Sales a WHERE customer.gender = 'F') a
INNER JOIN
  Transactions b
ON
  (CAST(a._id AS string) = CAST(b._id AS string));
```
Added spacing

File: docs/overview.md [hunk 9]
```diff
@@ -0,0 +1,844 @@
...
+Supported Joins: INNER JOIN, (CROSS) JOIN, LEFT OUTER JOIN, and RIGHT OUTER JOIN.
+
+### UNION ALL
+
+Combine result sets using UNION ALL:
+
+```sql
+SELECT *FROM Sales
+UNION ALL
+SELECT* FROM Transactions;
```
Comment (bucaojit) [inline docs/overview.md:360]:
```suggestion
SELECT * FROM Sales
UNION ALL
SELECT * FROM Transactions;
```

File: docs/overview.md [hunk 10]
```diff
@@ -0,0 +1,844 @@
...
+
+Note: UNION (which removes duplicates) is not supported. Only UNION ALL is supported.
+
+### Nested Selects
+
+Use subqueries with aliases:
+
+```sql
+SELECT *
+FROM (SELECT* FROM Sales) AS subSelect;
```
Comment (bucaojit) [inline docs/overview.md:371]:
```suggestion
FROM (SELECT * FROM Sales) AS subSelect;
```

File: docs/overview.md [hunk 11]
```diff
@@ -0,0 +1,844 @@
...
+
+The MongoDB Atlas SQL ODBC driver supports all authentication mechanisms
+supported by MongoDB (x509, OAuth, LDAP, etc...). See [authentication mechanisms](https://www.mongodb.com/docs/manual/core/authentication/#authentication-mechanisms)
+for a full list of supported authentication mechanisms.
+
+If [configured](https://www.mongodb.com/docs/manual/tutorial/configure-ssl/), the MongoDB Atlas SQL ODBC driver supports TLS/SSL connections.
+
+### Logging and Diagnostics
+
+By default, the MongoDB Atlas SQL ODBC driver produces logs in `/%HOME%/MongoDB/Atlas SQL/logs`.
```
Comment (bucaojit) [inline docs/overview.md:791]: This is for linux.  On windows it looks like we try to set it to:
```
C:\Users\{user}\Documents\MongoDB\Atlas SQL ODBC\{driver_log_version}\logs
```
Would be useful to add.

File: docs/overview.md [hunk 12]
```diff
@@ -0,0 +1,844 @@
...
+- DEBUG - Debug information useful for debegging purposes. Enable this mode to submit a log with a HELP ticket.
+- TRACE - Extremely verbose logging, including network traffic and information from ancillary libraries used in the driver. Not recommended unless
+asked for by MongoDB support.
+
+Each LOGLEVEL will include log messages emitted at a higher precedence. For example, INFO will include all INFO, WARN, and ERROR
+messages, while ERROR will include only ERROR messages.
+
+## Troubleshooting
+
+### Common Issues
```
Comment (bucaojit) [inline docs/overview.md:809]: We can add this message:

> Enterprise edition detected, but mongosqltranslate library not found.

Ensure that the `mongosqltranslate` library exists in the same directory as the MongoDB SQL ODBC driver.

**Review by bucaojit — APPROVED** [review]
Very thorough nice job! ✍️

**Pull request merged**